File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/WindowPropertyPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record WindowPropertyPacket(short windowId, short property, short value) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeByte(windowId);
        os.writeShort(property);
        os.writeShort(value);
    }

    @Override
    public byte getId() {
        return 0x31;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/WindowPropertyPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class WindowPropertyPacket(val windowId: Short, val property: Short, val value: Short) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeByte(windowId.toInt())
        os.writeShort(property.toInt())
        os.writeShort(value.toInt())
    }

    override fun getId(): Byte {
        return 0x31
    }
}
